tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/03_displayUnavailableProductAttributes.js

```diff
@@ -0,0 +1,156 @@
+    searchResultsPage: new SearchResultsPage(page),
+  };
+};
+
+describe('Enable/Disable display unavailable product attributes on the product page', async () => {
+  // before and after functions
+  before(async function () {
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    this.pageObjects = await init();
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+  });
+
+  // Login into BO and go to products page
+  loginCommon.loginBO();
+
+  it('should go to \'Catalog > Products\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToProductsPage', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.catalogParentLink,
+      this.pageObjects.boBasePage.productsLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
+    const pageTitle = await this.pageObjects.productsPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.productsPage.pageTitle);
+  });
+
+  it('should go to create product page and create a product', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'createProduct', baseContext);
+    await this.pageObjects.productsPage.goToAddProductPage();
+    const validationMessage = await this.pageObjects.addProductPage.createEditProduct(productData);
+    await expect(validationMessage).to.equal(this.pageObjects.addProductPage.settingUpdatedMessage);
+  });
+
+  it('should go to \'Shop parameters > Product Settings\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToProductSettingsPage', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.shopParametersParentLink,
+      this.pageObjects.boBasePage.productSettingsLink,
+    );
+    const pageTitle = await this.pageObjects.productSettingsPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.productSettingsPage.pageTitle);
+  });
+
+  const tests = [
+    {args: {action: 'disable', enable: false}},
+    {args: {action: 'enable', enable: true}},
+  ];
+  tests.forEach((test) => {
+    it(`should ${test.args.action} Display unavailable product attributes on the product page`, async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        `${test.args.action}DisplayUnavailableProductAttributes`,
+        baseContext,
+      );
+      const result = await this.pageObjects.productSettingsPage.setDisplayUnavailableProductAttributes(
+        test.args.enable,
+      );
+      await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+    });
+
+    it('should check the unavailable product attributes in FO product page', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        `checkUnavailableAttribute_${test.args.action}`,
```

Comment:
```suggestion
        `checkUnavailableAttribute${test.args.action[0].toUpperCase()}${test.args.action.slice(1)}`,
```
Or create a function for doing a uppercase on the first character. But don't mix snake case and camel case